### PR TITLE
Login and Main Page

### PR DESCRIPTION
**Pull request by sunshuh2**
The updates made in this pull request deals with the login, main page and also create project.

**TODO list for this part:**
1, Login now works with an example collection (login_test_project) in our csc207-project database, this example is my assumption for what a collection for project look like, changes could be required after we have confirmed how we are storing each part of the project in the database.
image
image
image
The images are parts for meeting/task/announcements in login_test_project, login currently works with this structure
2, Action Listeners for each function button in MainPageView.java needs to be implemented, the MainPageView construction related parts and function buttons may also need changes according to the functions we have.

**Improvements to make on currently running parts:**
1, The current code has a setPreferredSize() method used, and due to the structure of CardLayout, we have all views to have this size which makes views other than main page to have size that is too large for them. We can solve this by deleting setPreferredSize() but this means that main page will not have all its features showed when switched to it and we will need to enlarge the view manually. This could be a possible point to improve.
2, For this code now, we have the order of the list return from getInfoList(project, "announcementInfo") to actually depend on  the alphabetical order of the announcement's name or title (the field name). However, it should actually be depending on the creationTime so that each list has the most recent ones on the end of the list, so it will need some change. (I had this as improvement because this has to do only with how information from the database is presented on the main page)

### CODE REVIEW

File: src/data_access/FirebaseAccessObject.java [hunk 1]
```diff
@@ -52,10 +51,52 @@ public void save(Project project) {
             ApiFuture<WriteResult> result = docRef.set(data);
     }
 
-    public Project getProject(String projectName) {
-        return projects.get(projectName);
+    public Project getProjectInfo(String projectName) {
```
Comment (wogus00) [inline src/data_access/FirebaseAccessObject.java:54]: For getting information, almost all use cases will have some sort of get method.
Maybe we can use facade design principle for all get methods?

**Review by Kosei1227 — COMMENTED** [review]
Hi, thank you for the change. I would like to ask about LoginUseCaseFactory.java.
According to the constructor, the forth parameter should be LoginDataAccessInterface loginDataAccessObject. However, when i look at the Main.java, we have         LoginView loginView = LoginUseCaseFactory.createLoginView(viewManagerModel,loginViewModel,mainPageViewModel, firebaseAccessObject).
So, when I run the main.java, i get the error with the difference of the actual and expected types of 4th parameter. How did you fix this issue? Or, did you see the error when you run the code?

**Comment by wogus00** [comment]
Screenshot 2023-11-18 at 6 33 03 PM

Just tested your code and it seems to be working!

**Comment by sunshuh2** [comment]
> Hi, thank you for the change. I would like to ask about LoginUseCaseFactory.java. According to the constructor, the forth parameter should be LoginDataAccessInterface loginDataAccessObject. However, when i look at the Main.java, we have LoginView loginView = LoginUseCaseFactory.createLoginView(viewManagerModel,loginViewModel,mainPageViewModel, firebaseAccessObject). So, when I run the main.java, i get the error with the difference of the actual and expected types of 4th parameter. How did you fix this issue? Or, did you see the error when you run the code?

Hi, thank you for your review. I actually didn't see this error when running, could you provide a screenshot for this error case? Also, I guess this is not a problem because we have the FirebaseAccessObject to be implementing LoginDataAccessInterface so that the FirebaseAccessObject is a subtype of LoginDataAccessInterface.

**Comment by sunshuh2** [comment]
> Screenshot 2023-11-18 at 6 33 03 PM
> Just tested your code and it seems to be working!
> 
> But i also get this error message on terminal (i'm not sure if it's critical or not) Screenshot 2023-11-18 at 6 39 34 PM

Hi, thank you for your review. For this error message, I guess it occurs because we do not have an "announcementInfo" / "taskInfo" / "meetingInfo" document in the collection that you are trying to login. I guess this could be solve if we consider to add blank documents "announcementInfo" / "taskInfo" / "meetingInfo" to the collection when we create a project or we could test whether these document exists when are getting these info lists. This is indeed a problem so I guess I'll fix this later.

File: src/data_access/FirebaseAccessObject.java (hunk 1 above)
Comment (sunshuh2) [reply src/data_access/FirebaseAccessObject.java:54]: Sure, I think this is possible.

**Comment by sunshuh2** [comment]
> > Screenshot 2023-11-18 at 6 33 03 PM
> > Just tested your code and it seems to be working!
> > But i also get this error message on terminal (i'm not sure if it's critical or not) Screenshot 2023-11-18 at 6 39 34 PM
> 
> Hi, thank you for your review. For this error message, I guess it occurs because we do not have an "announcementInfo" / "taskInfo" / "meetingInfo" document in the collection that you are trying to login. We could solve this by considering adding blank documents "announcementInfo" / "taskInfo" / "meetingInfo" to the collection when we create a project or we could test whether these document exists when trying to get these info lists. This is indeed a problem so I guess I'll fix this later.

**Comment by Kosei1227** [comment]
> Hi, thank you for your review. I actually didn't see this error when running, could you provide a screenshot for this error case? Also, I guess this is not a problem because we have the FirebaseAccessObject to be implementing LoginDataAccessInterface so that the FirebaseAccessObject is a subtype of LoginDataAccessInterface.

I could fix the error. The issue was my pull from the main branch does not reflect some changes on FirebaseDataObject.
Thank you for providing you insight.

**Review by wogus00 — APPROVED** [review]
Tested that everything is working with all functionalities. Ready to be merged

**Pull request merged**